default/AI/ShipDesignAI.py

```diff
@@ -790,6 +816,102 @@ def update_stats(self, ignore_species=False):
             if self.attacks:
                 self.attacks = foAI.foAIstate.weight_attacks(self.attacks, weapons_grade)
 
+    def _apply_hardcoded_effects(self):
+        """Update stats that can not be read out by the AI yet, i.e. applied by effects.
+
+        This function should contain *all* hardcoded effects for hulls/parts to be considered by the AI
+        to make sure we can easily adjust the values for future balance changes or after implementing a
+        method to read out all stats.
+        """
```

Comment:
That would be recommended if it is a simple change in numbers and concerning effects you already find an example for.

Aside from maybe excessive changes in the organic growth, these stats generally shouldn't affect the AI's play by much though so the danger to "break" the AI is low. Most probably it is not even noticeable. If it is, then it should only be a very minor performance drop at this point.